fix warnings:

auth_sasldb.c: In function ‘auth_sasldb’:
auth_sasldb.c:144: warning: implicit declaration of function ‘gethostname’

auth_sasldb.c:153: warning: passing argument 8 of ‘_sasldb_getdata’ from incompatible pointer type
../sasldb/sasldb.h:60: note: expected ‘size_t *’ but argument is of type ‘int *’

--- saslauthd/auth_sasldb.c
+++ saslauthd/auth_sasldb.c
@@ -41,6 +41,7 @@
 #include <string.h>
 #include <stdlib.h>
 #include <pwd.h>
+#include <unistd.h>
 /* END PUBLIC DEPENDENCIES */
 
 #define RETURN(x) return strdup(x)
@@ -131,7 +132,8 @@
     /* VARIABLES */
     char pw[1024];			/* pointer to passwd file entry */
     sasl_utils_t utils;
-    int ret, outsize;
+    int ret;
+    size_t outsize;
     const char *use_realm;
     char realm_buf[MAXHOSTNAMELEN];
     /* END VARIABLES */
